evcache-core/src/main/java/com/netflix/evcache/EVCacheKey.java

```diff
@@ -99,16 +113,6 @@ public boolean equals(Object obj) {
                 return false;
         } else if (!canonicalKeyForDuet.equals(other.canonicalKeyForDuet))
             return false;
-        if (hashKey == null) {
-            if (other.hashKey != null)
-                return false;
-        } else if (!hashKey.equals(other.hashKey))
-            return false;
-        if (hashKeyForDuet == null) {
-            if (other.hashKeyForDuet != null)
-                return false;
-        } else if (!hashKeyForDuet.equals(other.hashKeyForDuet))
-            return false;
```

Comment:
@smadappa I haven't added any equal checks on the new hashmaps that contain hashed keys for duet and non-duet, due to the additional cost of such comparison. Consequently, we are now not considering hashkeys for equal check of EVCacheKey. Let me know if you see any concerns with this.